src/m365/base/ToDoCommand.spec.ts

```diff
@@ -0,0 +1,46 @@
+import assert from 'assert';
+import sinon from 'sinon';
+import auth from '../../Auth.js';
+import { telemetry } from '../../telemetry.js';
+import PowerAutomateCommand from './PowerAutomateCommand.js';
+import { accessToken } from '../../utils/accessToken.js';
+
+class MockCommand extends PowerAutomateCommand {
+  public get name(): string {
+    return 'mock';
+  }
+
+  public get description(): string {
+    return 'Mock command';
+  }
+
+  public async commandAction(): Promise<void> {
+  }
+
+  public commandHelp(): void {
+  }
+}
+
+describe('ToDoCommand', () => {
+  const cmd = new MockCommand();
+
+  before(() => {
+    sinon.stub(telemetry, 'trackEvent').returns();
+    auth.connection.accessTokens[auth.defaultResource] = {
+      expiresOn: 'abc',
+      accessToken: 'abc'
+    };
+    auth.connection.active = true;
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.connection.active = false;
+  });
+
+  it('throws error when trying to use the command using application only permissions', () => {
+    sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(true);
+    auth.connection.active = true;
+    assert.throws(() => (cmd as any).initAction({ options: {} }, {}));
```

Comment:
Let's check for the exact error message. Because now, any fail will make the test succeed. Let's fix this in the other tests as well.